### ISSUE TITLE
Logger creation fails if there is no network connection available

### ISSUE THREAD

**Issue by molszews** [whir1/serilog-sinks-graylog#12]
There is an issue when logger creation is called and the VM is also starting up, there doesn't seem to be network connectivity so the initial dns resolve throws an exception and the service doesn't start. 

```
Application: NotificationService.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Net.Sockets.SocketException
   at System.Net.Dns.HostResolutionEndHelper(System.IAsyncResult)
   at System.Net.Dns.EndGetHostAddresses(System.IAsyncResult)
   at System.Threading.Tasks.TaskFactory`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].FromAsyncCoreLogic(System.IAsyncResult, System.Func`2<System.IAsyncResult,System.__Canon>, System.Action`1<System.IAsyncResult>, System.Threading.Tasks.Task`1<System.__Canon>, Boolean)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at Serilog.Sinks.Graylog.Transport.DnsWrapper+<GetHostAddresses>d__0.MoveNext()

Exception Info: System.AggregateException
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean)
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].GetResultCore(Boolean)
   at System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].get_Result()
   at Serilog.Sinks.Graylog.GraylogSink..ctor(Serilog.Sinks.Graylog.GraylogSinkOptions)
   at Serilog.Sinks.Graylog.LoggerConfigurationGrayLogExtensions.Graylog(Serilog.Configuration.LoggerSinkConfiguration, Serilog.Sinks.Graylog.GraylogSinkOptions)
   at NotificationService.Infrastructure.Logging.SerilogInitializer.Initialize()
   at NotificationService.Program.Main(System.String[])
```

**Comment by daningalla** [comment]
Hi @molszews, how do you propose the sink should handle the failure?

**Comment by molszews** [comment]
I think the ultimate solution would be to have no network dependent logic in the sink constructor.
When logging event comes in, I will see if the dns name can be/is already resolved, if so I can log something, otherwise we'll try again with the next logging event.
I think that would be something similar to the way http sink works.

**Comment by daningalla** [comment]
Looking at a few other sinks, the common way to handle these things is to write an event to SelfLog and then consume the exception, thus failing silently. I'll get started on a P/R-hopefully whir1 approves and merges the other branches soon.

**Comment by j-schumann** [comment]
Hello,
are there any news on this issue? Having the same problem (System.Net.Sockets.SocketException (11001): Unknown host).

Serilog wiki explicitly requires the sinks to handle these errors: https://github.com/serilog/serilog/wiki/Reliability

> Sink authors: the responsibility for implementing this is on the sink implementation itself, so needs to be explicitly considered/addressed.

Thanks in advance for any PR!

**Comment by whir1** [comment]
fixed in version 2.0.5
